docs/user/datastores/query_interceptor.rst

```diff
@@ -0,0 +1,114 @@
+        */
+      def init(ds: DataStore, sft: SimpleFeatureType): Unit
+
+      /**
+        * Modifies the query in place
+        *
+        * @param query query
+        */
+      def rewrite(query: Query): Unit
+
+      /**
+       * Hook to allow interception of a query after extracting the query values
+       *
+       * @param strategy query strategy
+       * @return an exception if the query should be stopped
+       */
+      def guard(strategy: QueryStrategy): Option[IllegalArgumentException] = None
+    }
+
+Interceptors must have a default, no-arg constructor. The interceptor lifecycle consists of:
+
+1. The instance is instantiated via reflection, using its default constructor
+#. The instance is initialized via the ``init`` method, passing in the data store containing the simple feature type
+#. ``rewrite`` and ``guard`` are called once for each query
+#. When the data store is disposed, the instance is cleaned up via the ``close`` method
+
+Interceptors will be invoked in the order they are declared in the user data. In order to see detailed information
+on the results of query interceptors, you can enable ``TRACE``-level logging on the class
+``org.locationtech.geomesa.index.planning.QueryRunner$``.
+
+Provided Query Guards
+^^^^^^^^^^^^^^^^^^^^^
+
+GeoMesa provides some basic query guards that will block overly broad queries (which can overwhelm the system).
+For additional controls, see ``geomesa.query.timeout`` and ``geomesa.scan.block-full-table`` in
+:ref:`geomesa_site_xml`.
+
+Temporal Query Guard
+++++++++++++++++++++
+
+The temporal query guard will block queries which exceed a maximum temporal duration.
+Any query which attempts to return a larger time period will be stopped. This will not affect queries against indices that do not have
+a temporal component (for example, feature ID and attribute queries), nor queries that do not have any temporal
+component at all (which can usually be blocked by ``geomesa.scan.block-full-table``).
+
+To enable the guard, add ``org.locationtech.geomesa.index.planning.guard.TemporalQueryGuard``
+to ``geomesa.query.interceptors`` as indicated above, and set the duration using ``geomesa.guard.temporal.max.duration``:
+
+.. code-block:: java
+
+    sft.getUserData().put("geomesa.query.interceptors",
+      "org.locationtech.geomesa.index.planning.guard.TemporalQueryGuard");
+    sft.getUserData().put("geomesa.guard.temporal.max.duration", "28 days");
+
+Graduated Query Guard
++++++++++++++++++++++
+
+The graduated query guard applies different duration limits based on the spatial extent of the query.
+As a query becomes larger in space, it can be limited to shorter and shorter time ranges.
+A series of rules limit the duration for queries which are at most a given size in square degrees.
+
+To enable the guard, add ``org.locationtech.geomesa.index.planning.guard.GraduatedQueryGuard``
+to ``geomesa.query.interceptors`` as indicated above.  Configuration is managed via TypeSafe config.
```

Comment:
should probably clarify here that standard config loading is used and the config is expected under the key `geomesa.guard.graduated`

Reply:
Adding some docs for that.